src/parquet/file/writer-internal.cc

```diff
@@ -142,7 +142,7 @@ ColumnWriter* RowGroupSerializer::NextColumn() {
   // Throws an error if more columns are being written
   auto col_meta = metadata_->NextColumnChunk();
 
-  if (current_column_writer_) { total_bytes_written_ = current_column_writer_->Close(); }
+  if (current_column_writer_) { total_bytes_written_ += current_column_writer_->Close(); }
```

Comment:
The tests you updated with multiple columns should be sufficient. However, we should keep this check until we feel confident. JIRA to remove the redundancy PARQUET-730 later.